### PR TITLE
feat(project_cleanup): remove clusters prior to project

### PR DESCRIPTION
**Pull request by apeabody**

### CODE REVIEW

File: modules/project_cleanup/function_source/main.go [hunk 1]
```diff
@@ -611,6 +624,33 @@ func invoke(ctx context.Context) {
...
+
+		logger.Printf("Got [%d] clusters for project [%s]", len(listResponse.Clusters), projectId)
+		if len(listResponse.Clusters) == 0 {
+			return 0
+		}
+
+		for _, cluster := range listResponse.Clusters {
+			logger.Printf("Try to remove cluster: %s", cluster.Name)
+
+			reqDCR := &containerpb.DeleteClusterRequest{Name: fmt.Sprintf("projects/%s/locations/%s/clusters/%s", projectId, cluster.Location, cluster.Name)}
```
Comment (bharathkkb) [inline modules/project_cleanup/function_source/main.go:644]: If a cluster is in stopping or any other state could we still request delete?

https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.Status

File: modules/project_cleanup/function_source/main.go [hunk 2]
```diff
@@ -611,6 +624,33 @@ func invoke(ctx context.Context) {
...
+			_, err := containerService.DeleteCluster(ctx, reqDCR)
+			if err != nil {
+				logger.Printf("Failed to delete cluster [%s] for [%s], error [%s]", cluster.Name, projectId, err.Error())
+			}
+		}
+
+		return len(listResponse.Clusters)
```
Comment (bharathkkb) [inline modules/project_cleanup/function_source/main.go:651]: Could there be a possibility that len > 0 but all clusters got an err response for `DeleteCluster` so it won't move forward? My thinking is we could return `len(listResponse.Clusters)-len(deleteFailures)`

**Review by bharathkkb — APPROVED** [review]
Thanks @apeabody!

File: modules/project_cleanup/function_source/main.go (hunk 1 above)
Comment (apeabody) [reply modules/project_cleanup/function_source/main.go:644]: Yes, good point - Deleting a `STOPPING` seemed to work, however if in `PROVISIONING` (although that status is unlikely during cleanup) it raised an error.  Let me add a switch based on the status as even for STOPPING we can make it more efficient.

File: modules/project_cleanup/function_source/main.go (hunk 2 above)
Comment (apeabody) [reply modules/project_cleanup/function_source/main.go]: Yeah - I decided to implement a new `pendingDeletion` int using the switch case to provide finer grain granularity.

**Review by bharathkkb — APPROVED** [review]

File: modules/project_cleanup/function_source/main.go (hunk 1 above)
Comment (bharathkkb) [reply modules/project_cleanup/function_source/main.go]: lgtm!

**Pull request merged**